### DISCUSSION TITLE
Need some help stacking UI widgets

### DISCUSSION THREAD

**Discussion by etaletovic** [koreader/koreader#8881] [Q&A]
I am experimenting with UI and trying to build a composite widget of Menu + something else. I tried putting Menu inside VerticalGroup but then I realized that page navigation buttons no longer work. I am hoping someone has any tips for stacking 2 widgets, one of them being Menu.

Here is the code reproducing the issue I have:

```
    local menu = Menu:new{
        width = Screen:getWidth(),
        height = Screen:getHeight(),
        is_borderless = true,
        _manager = self,
        no_title = true,
        is_popout = true
    }

    local items = {}
    for _ = 1, 100 do
        table.insert(items, { text = "Test item" })
    end

    menu:switchItemTable("", items)

    UIManager:show(
        VerticalGroup:new{
            menu
        }
    )
```

![image](https://user-images.githubusercontent.com/5982655/157184309-b8dcfed4-d160-4e40-bea9-aca293b7500a.png)

If I take out VerticalGroup then page turning works again
```
 UIManager:show(menu)
```

![image](https://user-images.githubusercontent.com/5982655/157184664-70702211-60aa-4b94-81ed-897fd9454c21.png)

**Comment by comphilip** [comment]
Maybe you missing setting menu's `show_parent` property.
```lua
local vg = VerticalGroup:new{
    menu
}
menu.show_parent = vg
UIManager:show(vg)
```

**Reply by etaletovic** [reply]
Ok this seems to work but I don't really understand what show_parent does exactly.

**Reply by NiLuJe** [reply]
Quick, shameless self plug: I've explained some of it in this wall of text: http://koreader.rocks/doc/modules/ui.uimanager.html#UIManager:setDirty :D

**Reply by Frenzie** [reply]
> Beware! This doesn't take any prisoners!

lol
